src/core/RandomVariables.jl

```diff
@@ -208,19 +745,79 @@ function Base.getindex(vi::AbstractVarInfo, vns::Vector{<:VarName})
         reconstruct(dist, getval(vi, vns), length(vns)))
 end
 
-# NOTE: vi[vview] will just return what insdie vi (no transformations applied)
-Base.getindex(vi::UntypedVarInfo, vview::VarView) = copy(getval(vi, vview))
-Base.setindex!(vi::UntypedVarInfo, val::Any, vview::VarView) = setval!(vi, val, vview)
+"""
+`getindex(vi::VarInfo, spl::Union{SampleFromPrior, Sampler})`
+
+Returns the current value(s) of the random variables sampled by `spl` in `vi`. The value(s) may or may not be transformed to Eucledian space.
+"""
+Base.getindex(vi::AbstractVarInfo, spl::SampleFromPrior) = copy(getall(vi))
+Base.getindex(vi::UntypedVarInfo, spl::Sampler) = copy(getval(vi, _getranges(vi, spl)))
+function Base.getindex(vi::TypedVarInfo, spl::Sampler)
+    # Gets the ranges as a NamedTuple
+    # getfield(ranges, f) is all the indices in `vals` of the `vn`s with symbol `f` sampled by `spl` in `vi`
```

Comment:
There is no `getfield(ranges, f)` any more.

Reply:
`ranges` refers to the output variable from `_getranges`. Calling `getfield(ranges, f)` is what this line refers to. I will try to make it clearer.